openmdao/core/driver.py

```diff
@@ -22,7 +22,80 @@
 from openmdao.utils.array_utils import sizes2offsets
 from openmdao.vectors.vector import _full_slice, _flat_full_indexer
 from openmdao.utils.indexer import indexer
-from openmdao.utils.om_warnings import issue_warning, DerivativesWarning, DriverWarning
+from openmdao.utils.om_warnings import issue_warning, DerivativesWarning, \
+    DriverWarning, OMDeprecationWarning
+
+
+class DriverResult():
+    """
+    A container that stores information pertaining to the result of a driver execution.
+
+    Attributes
+    ----------
+    runtime : float
+        The time required to execute the driver, in seconds.
+    iter_count : int
+        The number of iterations used by the optimizer.
+    obj_calls : int
+        The number of times the objective function was evaluated (model solve_nonlinear calls).
+    deriv_calls : int
+        The number of times the total jacobian was computed.
+    exit_status : str
+        A string that may provide more detail about the results of the driver run.
+    success : bool
+        A boolean that dictates whether or not the driver was successful.
+    """
+
+    def __init__(self):
+        """
+        Initialize the DriverResult object.
+        """
+        self.runtime = 0.0
+        self.iter_count = 0
+        self.obj_calls = 0
+        self.deriv_calls = 0
+        self.exit_status = 'NOT_RUN'
```

Comment:
In Aviary and MBSE, we have been using python enums for things like this.  It makes it easer for someone coding a new driver, so that they can just import the enum and select one of the exit statuses instead of accidentally making up a new string. (Of course, if they really do have a new kind of status, they can add that to the enum.)

Reply:
Hmm, this isn't a new addition though. This is someothing that's alread existent that the optimization report relies upon.